packages/amplication-client/src/Resource/ResourceHome.tsx

```diff
@@ -50,7 +114,60 @@ const ResourceHome = ({ match, innerRoutes }: Props) => {
                 resourceThemeMap[currentResource?.resourceType].color,
             }}
           >
-            {currentResource?.name}
+            <div className={`${CLASS_NAME}__header__input__container`}>
+              <Formik
+                initialValues={{ name: currentResource?.name }}
+                validate={(values: models.Resource) =>
+                  validate(values, FORM_SCHEMA)
+                }
+                enableReinitialize
+                onSubmit={handleSubmit}
+              >
+                {({ errors, isValid }) => {
+                  return (
+                    <div className={`${CLASS_NAME}__header__form`}>
+                      {isEditing ? (
+                        <Form>
+                          <FormikAutoSave debounceMS={1000} />
+                          <Field
+                            autoFocus={true}
+                            className={`${CLASS_NAME}__header__input`}
+                            name="name"
+                            as="input"
+                            onBlur={() => handleBlur(isValid)}
+                          />
+                          {isValid ? (
+                            showTick && (
+                              <Icon
+                                className={`${CLASS_NAME}__header__saved`}
+                                icon={"check"}
+                                size="medium"
+                              />
+                            )
+                          ) : (
+                            <Icon
+                              icon="info_circle"
+                              size="small"
+                              className={`${CLASS_NAME}__invalid`}
+                            />
+                          )}
+                        </Form>
+                      ) : (
+                        <span
+                          className={`${CLASS_NAME}__header__text`}
+                          onClick={() => setIsEditing(true)}
```

Comment:
please use useCallback to define the function

Reply:
Can you be specific on using useCallback on which function, please?